tests/test_utils.py

```diff
@@ -174,5 +183,23 @@ def test_read_yaml_string_with_env_var_not_exist():
     user: ${USER_NAME}
     password: ${PASSWORD}
     """
-    with pytest.raises(KeyError):
+    with pytest.raises(ValueError) as error_info:
         r = utils.read_yaml_string(config_with_env_var_not_exist)
+    error_message = ("Error when trying to expand the environment variables "
```

Comment:
I think it's ok when we simply test the exception. Testing for the actual log message seems a bit too much for me.